GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m

```diff
@@ -1301,4 +1312,60 @@ - (void)testAppDelegateIsProxiedIncludingAPNSMethodsWhenEnabled {
   XCTAssertNotEqualObjects([originalAppDelegate class], originalAppDelegateClass);
 }
 
+#pragma mark - Test UISceneDelegate proxy
+
+#if ((TARGET_OS_IOS || TARGET_OS_TV) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= 130000))
+
+- (void)testProxySceneDelegate {
+  if (@available(iOS 13, tvOS 13, *)) {
+    GULTestSceneDelegate *realSceneDelegate = [[GULTestSceneDelegate alloc] init];
+    id mockSharedScene = OCMClassMock([UIScene class]);
+    OCMStub([mockSharedScene delegate]).andReturn(realSceneDelegate);
+    size_t sizeBefore = class_getInstanceSize([GULTestSceneDelegate class]);
+
+    Class realSceneDelegateClassBefore = [realSceneDelegate class];
+
+    [GULAppDelegateSwizzler proxySceneDelegate:mockSharedScene];
+
+    XCTAssertTrue([realSceneDelegate isKindOfClass:[GULTestSceneDelegate class]]);
+
+    NSString *newClassName = NSStringFromClass([realSceneDelegate class]);
+    XCTAssertTrue([newClassName hasPrefix:@"GUL_"]);
+    // It is no longer GULTestSceneDelegate class instance.
+    XCTAssertFalse([realSceneDelegate isMemberOfClass:[GULTestSceneDelegate class]]);
+
+    size_t sizeAfter = class_getInstanceSize([realSceneDelegate class]);
+
+    // Class size must stay the same.
+    XCTAssertEqual(sizeBefore, sizeAfter);
+
+    // After being proxied, it should be able to respond to the required method selector.
+    XCTAssertTrue([realSceneDelegate respondsToSelector:@selector(scene:openURLContexts:)]);
+
+    // Make sure that the class has changed.
+    XCTAssertNotEqualObjects([realSceneDelegate class], realSceneDelegateClassBefore);
+  }
+}
+
+- (void)testProxyProxiedSceneDelegate {
+  if (@available(iOS 13, tvOS 13, *)) {
+    GULTestSceneDelegate *realSceneDelegate = [[GULTestSceneDelegate alloc] init];
+    id mockSharedScene = OCMClassMock([UIScene class]);
+    OCMStub([mockSharedScene delegate]).andReturn(realSceneDelegate);
+
+    // Proxy the scene delegate for the 1st time.
+    [GULAppDelegateSwizzler proxySceneDelegate:mockSharedScene];
+
+    Class realSceneDelegateClassBefore = [realSceneDelegate class];
+
+    // Proxy the scene delegate for the 2nd time.
+    [GULAppDelegateSwizzler proxySceneDelegate:mockSharedScene];
+
+    // Make sure that the class isn't changed.
+    XCTAssertEqualObjects([realSceneDelegate class], realSceneDelegateClassBefore);
+  }
+}
+
```

Comment:
Could you also add tests to make sure that interseptors are notified when the swizzled methods are called (similar to [this one](https://github.com/firebase/firebase-ios-sdk/blob/b05d8d186ee41cc5ec0f83967b865db10e7f1f41/GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m#L1042), please.

Reply:
Done.